### PR TITLE
Improve error messages

### LINKED ISSUE #28: Print out key giving ValidationError
The [jsonschema](https://pypi.org/project/jsonschema/) package only prints out the value giving a `ValidationError`, not the associated key.  This is often not informative, for example, if the value is `None`:
```
error - None is not of type 'string' in None
```
arising from a `submission.yaml` file containing:
```yaml
data_license: {description: null, name: null, url: null}
```
This line was originally written by default when converting from the `oldhepdata` format (see HEPData/hepdata-converter#32), therefore many old submissions are now invalidated.  The key giving rise to a `ValidationError` should be printed in addition to the invalid value.  See [Handling Validation Errors](https://python-jsonschema.readthedocs.io/en/stable/errors/#handling-validation-errors) in the `jsonschema` docs.

### PR DESCRIPTION
**Pull request by alisonrclarke**
Fixes #28.

### CODE REVIEW

File: testsuite/test_data/invalid_submission_license.yaml [hunk 1]
```diff
@@ -0,0 +1,69 @@
+---
+
+additional_resources:
+  - location: "http:github.com/HEPData/hepdata"
+    description: "Full source code for creating this data"
+  - location: "http:github.com/HEPData/hepdata"
+    description: "Full source code for creating this data"
```
Comment (GraemeWatt) [inline testsuite/test_data/invalid_submission_license.yaml:7]: All the test data has this invalid URL `http:github.com/HEPData/hepdata`.  Maybe correct it to `https://github.com/HEPData/hepdata`?

**Review by GraemeWatt — CHANGES_REQUESTED** [review]
Looks good.  Can you please make a similar improvement for the `ValidationError` raised by the `DataFileValidator` class?

File: testsuite/test_data/invalid_submission_license.yaml (hunk 1 above)
Comment (alisonrclarke) [reply testsuite/test_data/invalid_submission_license.yaml:7]: Will do. I copied it from another test file - looks like there are lots of instances of this so I'll fix all of them (provided that's not the point of a test!).

**Review by GraemeWatt — APPROVED** [review]
Thanks for the improvements.  A couple of tests were failing on my laptop with PyYAML v5.3.1, so I've specified a minimum version of 5.4.1.  I've also specified `python_requires='>=3.6'` to prevent installation using Python 2.  I'll merge now.

**Pull request merged**